Sources/SwiftParser/Diagnostics/Utils.swift

```diff
@@ -22,3 +22,14 @@ extension String {
     return String(result)
   }
 }
+
+extension Collection {
+  /// If the collection contains a single element, return it, otherwise `nil`.
+  var only: Element? {
+    if self.count == 1 {
+      return self.first!
+    } else {
+      return nil
+    }
```

Comment:
Ah yes, another `.only`. `if !isEmpty && index(after: startIndex) == endIndex` could be slightly better than doing a count for some collections (I think we discuss this every time there's a new `.only` 😆), though for all the ones we'll use internally here `count` is fine 🤷

Reply:
Yes, it just happens to always be useful in all sorts of contexts… Good suggestion with the performance improvement though. I haven’t been part of any of these discussions yet. 😆

Reply:
It would only matter in cases where `count` is actually expensive, so not here. But thought I'd mention anyway :).